course7/homeworks/anton/dc/main.cc

```diff
@@ -1,19 +1,21 @@
 #include "complex.h"
 #include "operation.h"
 #include "parser.h"
+#include "my_linked_list.h"
 #include <boost/filesystem.hpp>
 #include <dlfcn.h>
 #include <map>
 #include <stack>
 
 namespace fs = boost::filesystem;
 
-using invoke_command = void (*)(std::stack<Complex>&);
+using Stack = std::stack<Complex, MyComplexLinkedList>;
+using invoke_command = void (*)(Stack&);
 using get_char = char (*)();
 
-std::ostream& operator<<(std::ostream& s, std::stack<Complex>& stack)
+std::ostream& operator<<(std::ostream& s, Stack& stack)
 {
-    std::stack<Complex> tmp_stack;
+    Stack tmp_stack;
     while(!stack.empty()) {
```

Comment:
Once you implemented the iterators, you should be able to just iterate over the stack here

Reply:
@maciekgajewski please could you elaborate? `std::stack` doesn't expose `begin` and `end` methods